OpenRA.Game/Traits/Player/Shroud.cs

```diff
@@ -172,18 +183,14 @@ void Invalidate(IEnumerable<PPos> changed)
 					type = count > 0 ? ShroudCellType.Visible : ShroudCellType.Fog;
 				}
 
+				oldResolvedType[uv] = resolvedType[uv];
 				resolvedType[uv] = type;
+				if (type != oldResolvedType[uv])
```

Comment:
This seems to be the only place we use `oldResolvedType`. Is there a reason to not just store `var oldType = resolvedType[uv];` and avoid the extra `CellLayer`?

Reply:
Must be a vestige of some earlier WIP. Removed.